dm/syncer/data_validator.go

```diff
@@ -549,6 +568,22 @@ func (v *DataValidator) doValidate() {
 				v.sendError(terror.ErrValidatorPersistData.Delegate(err))
 				return
 			}
+		case *replication.QueryEvent:
+			locationForFlush.Position = currLoc.Position
+			query := string(ev.Query)
+			if query == "COMMIT" || query == "BEGIN" {
+				break
+			}
+			err = locationForFlush.SetGTID(ev.GSet)
+			if err != nil {
+				v.sendError(terror.Annotate(err, "failed to set gtid"))
+				return
+			}
+			if err = v.persistCheckpointAndData(locationForFlush); err != nil {
```

Comment:
why this is persistCheckpointAndData but other branches are checkAndPersistCheckpointAndData

Reply:
to make sure checkpoint is persisted on ddl, so i t can check it. but seems we can use checkAndPersistCheckpointAndData and set a very small interval.